packages/xdl/src/project/getPublishExpConfigAsync.ts

```diff
@@ -30,16 +30,26 @@ export async function getPublishExpConfigAsync(
   options.releaseChannel = options.releaseChannel || 'default';
 
   // Verify that exp/app.json and package.json exist
-  const { exp: privateExp } = getConfig(projectRoot);
-  const { hooks } = privateExp;
-  const { exp, pkg } = getConfig(projectRoot, { isPublicConfig: true });
-
+  const {
+    exp: { hooks, runtimeVersion },
+  } = getConfig(projectRoot, { skipSDKVersionRequirement: true });
+  const { exp, pkg } = getConfig(projectRoot, {
+    isPublicConfig: true,
+    // enforce sdk validation if user is not using runtimeVersion
+    skipSDKVersionRequirement: !!runtimeVersion,
+  });
   const { sdkVersion } = exp;
+  if (!sdkVersion) {
+    throw new Error(
+      'Config is missing an SDK version. See https://docs.expo.io/bare/installing-updates/'
+    );
+  }
```

Comment:
This is included as although we no longer enforce SDK validation when using runtimeVersion, we still need the SDK version to be defined in order to prevent various expo libraries that require the SDK version to be defined from crashing.

Reply:
i don't think any libraries require sdk version to be defined - what are some examples you had in mind?

Reply:
OK removed after slack convo.